static/src/javascripts/bootstraps/enhanced/liveblog.js

```diff
@@ -94,6 +99,30 @@ const initTracking = () => {
 			ophan.record(componentEvent(pinnedBlockId, 'VIEW'));
 		});
 	}
+
+	// pinned post duration tracking
+    let hasBeenSeen = false;
+    const pinnedPostTiming = measureTiming('pinned-post-view-duration');
+    const originalPinnedBlockId = pinnedBlock.dataset.blockId
+    const onIntersect = (entries) => {
+        entries
+            .forEach((entry) => {
+                if (entry.isIntersecting) {
+                    hasBeenSeen = true;
+                    pinnedPostTiming.clear();
+                    pinnedPostTiming.start();
+                } else if (hasBeenSeen) {
+                    const timeTaken = pinnedPostTiming.end();
+                    if (timeTaken) {
+                        trackOphanPinnedDuration(originalPinnedBlockId, timeTaken);
+                    }
+                }
+            });
+    };
+    const observer = new IntersectionObserver(onIntersect, {
+        threshold: 0.8,
+    });
+    observer.observe(pinnedBlock);
 };
```

Comment:
I'm wary of abstracting too much but I wonder if it might be nice to have this in its own function?